systemtests/bin/make-stacks.py

```diff
@@ -56,22 +56,30 @@ def build_stack(data):
         return {}
 
     modules = []
+    modules_list = []
     for module in json_dump.get("modules", []):
         debug_file = module.get("debug_file", "")
         debug_id = module.get("debug_id", "")
+        if not debug_file or not debug_id:
+            continue
+        # Add the module information to the map
         modules.append((debug_file, debug_id))
-
-    modules_list = [m["filename"] for m in json_dump.get("modules", [])]
+        # Keep track of which modules are at which index
+        modules_list.append(module["filename"])
```

Comment:
We need to make sure these two lists are in lock step because we're potentially skipping modules.